e2e/tests/flag_test.go

```diff
@@ -94,3 +98,79 @@ func TestFlagDenyDomain(t *testing.T) {
 	newClient(t, "https://www.google.com").GET("/").ExpectStatus(http.StatusForbidden)
 	newClient(t, httpbin).GET("/status/200").ExpectStatus(http.StatusOK)
 }
+
+func TestFlagReadLimit(t *testing.T) {
+	const (
+		// It streams 2 * 5MiB, the read limit is 1MiB/s, but minimum burst is 4MiB, so it should take approximately 6 seconds.
+		connections     = 2
+		size            = 5 * 1024 * 1024 // 5MiB
+		expectedTime    = 6 * time.Second
+		acceptableRange = expectedTime * 5 / 100 // 5%
+	)
+	var wg sync.WaitGroup
+
+	ts := time.Now()
+	for i := 0; i < connections; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			c := newClient(t, httpbin)
+			c.GET(fmt.Sprintf("/stream-bytes/%d", size)).ExpectStatus(http.StatusOK).ExpectBodySize(size)
+		}()
+	}
+	wg.Wait()
+	if elapsed := time.Since(ts); elapsed < expectedTime-acceptableRange || elapsed > expectedTime+acceptableRange {
+		t.Fatalf("Expected request to take approximately %s, took %s", expectedTime, elapsed)
+	}
+}
+
+func TestFlagWriteLimit(t *testing.T) {
+	const (
+		// It streams 2 * 5MiB, the write limit is 1MiB/s, but minimum burst is 4MiB, so it should take approximately 6 seconds.
+		connections     = 2
+		size            = 5 * 1024 * 1024 // 5MiB
+		expectedTime    = 6 * time.Second
+		acceptableRange = expectedTime * 5 / 100 // 5%
+	)
+	var wg sync.WaitGroup
+
+	ts := time.Now()
+	for i := 0; i < connections; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			c := newClient(t, httpbin)
+			c.GET("/count-bytes/", func(req *http.Request) {
+				req.Header.Set("Content-Type", "application/octet-stream")
+				req.Body = &bytesReaderCloser{bytes.NewReader(make([]byte, size))}
+			}).ExpectStatus(http.StatusOK).ExpectHeader("body-size", fmt.Sprintf("%d", size))
+		}()
+	}
+	wg.Wait()
+	if elapsed := time.Since(ts); elapsed < expectedTime-acceptableRange || elapsed > expectedTime+acceptableRange {
+		t.Fatalf("Expected request to take approximately %s, took %s", expectedTime, elapsed)
+	}
+}
+
+func TestFlagReadWriteLimit(t *testing.T) {
```

Comment:
I do not see why we would want to run it twice.

Reply:
What do we run twice? It runs Read and Write test concurrently.